envs/monkey_zoo/blackbox/utils/credential_config_generator/credential_config_generator.py

```diff
@@ -0,0 +1,98 @@
+import argparse
+import copy
+import json
+import os
+from pathlib import Path
+
+import nanoid as nanoid
+
+parser = argparse.ArgumentParser(
+    prog="Credential Config Generator",
+    description="Small script to add usernames and " "passwords to Infection Monkey configuration",
+)
+
+parser.add_argument(
+    "-u",
+    "--userlist-path",
+    type=str,
+    required=True,
+    help="Path to userlist file. Userfile should contain one username per line.",
+)
+parser.add_argument(
+    "-p",
+    "--passwordlist-path",
+    type=str,
+    required=True,
+    help="Path to passwordlist file. " "Passwordfile should contain one password per line.",
+)
+parser.add_argument(
+    "-c",
+    "--config-path",
+    type=str,
+    help="Path to Infection Monkey configuration file."
+    "Must be unencrypted. If not provided, will use default config from a file.",
+)
+
+output_path = (
+    Path(os.path.dirname(os.path.realpath(__file__))) / "generated_configs" / "generated.conf"
+)
+default_config_path = Path(os.path.dirname(os.path.realpath(__file__))) / "default.conf"
+
+args = parser.parse_args()
+
+empty_credential = {
+    # NanoId with 21 characters
+    "id": "",
+    "identity": "",
+    "password": "",
+    "lm": "",
+    "ntlm": "",
+    "ssh_public_key": "",
+    "ssh_private_key": "",
+    "isNew": True,
+}
```

Comment:
Isn't the config generated here going to be sent to the propagation credentials endpoint directly?

Reply:
Yes, during configuration import credentials are sent to the endpoint

Reply:
Then the format of this should be `{"identity": "", "secret": {...}}`. That's what the endpoint expects. The format here right now is how the frontend parses it.